main/src/main/scala/sbt/CommandProgress.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * sbt
+ * Copyright 2023, Scala center
+ * Copyright 2011 - 2022, Lightbend, Inc.
+ * Copyright 2008 - 2010, Mark Harrah
+ * Licensed under Apache License 2.0 (see LICENSE)
+ */
+
+package sbt
+
+/**
+ * Tracks command execution progress. In addition to ExecuteProgress, this interface
+ * adds command start and end events, and gives access to the sbt.State at the beginning
+ * and end of each command.
+ */
+trait CommandProgress extends ExecuteProgress[Task] {
```

Comment:
Along the same line as what Adrien said, I wonder if this should be called ExecuteProgrogress2. I did something similar recently on the Zinc side and added an interface called AnalysisCallback2.